src/wallet/wallet.cpp

```diff
@@ -2215,6 +2215,26 @@ void CWallet::AvailableCoins(std::vector<COutput> &vCoins, bool fOnlySafe, const
                     continue;
                 }
 
+
+                if (coinControl) {
```

Comment:
Correct me if I'm wrong but this block will incur in a substantial performance penalty? What about:
```cpp
if (coinControl && coinControler->m_input_type != InputType::ALL) {
```